projects/ui/src/components/Common/Form/BorderEffect.tsx

```diff
@@ -22,31 +22,23 @@ const BorderEffect: FC<
   const actions = useMemo(() => {
     const handleMouseOver = () => {
       if (!ref.current || activeRef.current || disabled) return;
-      ref.current.style.padding = '1px';
-      ref.current.style.border = '1px solid';
-      ref.current.style.borderColor = BeanstalkPalette.textBlue;
+      ref.current.style.boxShadow = `inset 0 0 0 1px ${BeanstalkPalette.textBlue}`;
     };
 
     const handleMouseLeave = () => {
       if (!ref.current || activeRef.current || disabled) return;
-      ref.current.style.padding = '1px';
-      ref.current.style.border = '1px solid';
-      ref.current.style.borderColor = BeanstalkPalette.inputGrey;
+      ref.current.style.boxShadow = `inset 0 0 0 1px ${BeanstalkPalette.inputGrey}`;
     };
 
     const handleOnClick = () => {
       if (!ref.current || activeRef.current || disabled) return;
-      ref.current.style.padding = '0px';
-      ref.current.style.border = '2px solid';
-      ref.current.style.borderColor = theme.palette.primary.main;
+      ref.current.style.boxShadow = `inset 0 0 0 2px ${theme.palette.primary.main}`;
       activeRef.current = true;
     };
 
     const handleClickAway = () => {
       if (!ref.current || !activeRef.current || disabled) return;
-      ref.current.style.padding = '1px';
-      ref.current.style.border = '1px solid';
-      ref.current.style.borderColor = BeanstalkPalette.inputGrey;
+      ref.current.style.boxShadow = `inset 0 0 0 1px ${BeanstalkPalette.inputGrey}`;
       activeRef.current = false;
     };
```

Comment:
Future work: This pattern is sufficient for now but let's explore aways to avoid using direct DOM modification for this, CSS will be cleaner & faster.